### PR TITLE
Modify AB3 Model Components

### PR DESCRIPTION
**Pull request by aweijun**
Modify the original address book class to fit Trackermon.

1. Create Name class to represent name of show.
2. Create Show class to represent shows within Trackermon
3. Create Tag class to represent tags of the show.
4. Create DuplicatedShowException is thrown when there is a duplicated shows.
5. Create ShowNotFoundException is thrown when no show can be found.
6. Create ReadOnlyShowList
7. Create ShowList
8. Create UniqueShowList
9. Create ModelManagerTrackermon
10. Create ModelTrackermon
11. Create ReadOnlyUserPrefsTrackermon
12. Create UserPrefsTrackermon

### CODE REVIEW

**Review by JonathanHoshi — APPROVED** [review]
Looks good to me as a base class for the team to work on!

**Review by arcornior — APPROVED** [review]
LGTM!

**Review by Ardentsoul — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/model/ModelTrackermon.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    GuiSettings getGuiSettings();
+
+    /**
+     * Sets the user prefs' GUI settings.
+     */
+    void setGuiSettings(GuiSettings guiSettings);
+
+    /**
+     * Returns the user prefs' address book file path.
```
Comment (lowkaiwei98) [inline src/main/java/seedu/address/model/ModelTrackermon.java:35]: I realized while using this code for the v1.3. Comments were not updated here. Will change it for you.

**Pull request merged**